test/controllers/sessions_controller_test.rb

```diff
@@ -254,5 +386,32 @@ def assert_redirected_to_top_level(shop_domain, expected_url = nil)
           elements.first["data-target"]
       end
     end
+
+    def assert_redirected_to_embedded(shop_domain, base_embedded_url = nil)
+      assert_not_nil base_embedded_url
+      redirect_uri = "https://test.host/login?shop=#{shop_domain}"
+      expected_url = base_embedded_url + "?redirectUri=#{CGI.escape(redirect_uri)}" + "&shop=#{shop_domain}"
+
+      assert_redirected_to(expected_url)
+    end
+
+    def setup_context
+      ShopifyAPI::Context.setup(
+        api_key: ShopifyApp.configuration.api_key,
+        api_secret_key: ShopifyApp.configuration.secret,
+        api_version: ShopifyApp.configuration.api_version,
+        host_name: "test.host",
+        scope: ShopifyApp.configuration.scope,
+        is_private: false,
+        is_embedded: ShopifyApp.configuration.embedded_app,
+        session_storage: ShopifyApp::SessionRepository,
+      )
+    end
+
+    def with_embedded_redirect_url
+      ShopifyApp.configuration.embedded_redirect_url = "/a-redirect-page"
+      yield
+      ShopifyApp.configuration.embedded_redirect_url = nil
```

Comment:
Should we just always set this value to `nil` in the `test_helper` instead of resetting it here?